skin/frontend/base/default/js/TIG/MyParcelBE/myparcel.js

```diff
@@ -32,438 +42,498 @@ MyParcel = {
+        mypajQuery('#mypa-input').val(JSON.stringify(result));
+        if (typeof window.mypa != 'undefined') {
+            window.mypa.fn.fnCheckout.saveShippingMethod();
         }
+    },
 
-        /* No pre selection, show everything. */
-        else {
-            MyParcel.showAllDeliveryOptions();
-            mypajQuery('#method-myparcel-flatrate').click();
-        }
+    addStyleToPrice: function (chosenDelivery) {
+        mypajQuery(chosenDelivery).addClass('mypa-bold-price');
     },
 
-    showPostNlPrices: function () {
-        var priceMap = {
-            "pricePostNLFlatrate": "mypa-postnl-flatrate-price",
-            "pricePostNLSignature": "mypa-postnl-signature-price",
-            "pricePostNLRecipientOnly": "mypa-postnl-recipient-only-price",
-            "pricePostNLRecipientOnlySignature": "mypa-postnl-recipient-only-signature-price",
-            "pricePostNLEvening": "mypa-postnl-evening-price",
-            "pricePostNLPickupExpresse": "mypa-postnl-pickup-express-price",
-            "pricePostNLMorning": "mypa-postnl-morning-price",
-            "pricePostNLMorningSignature": "mypa-postnl-morning-signature-price",
+    removeStyleFromPrice: function (){
+        mypajQuery('.mypa-delivery-option-table').find( "span" ).removeClass('mypa-bold-price');
+    },
 
-            "pricePostNLEveningSignature": "mypa-postnl-evening-signature-price",
+    addDeliveryToMagentoInput: function (deliveryMomentOfDay) {
 
-            "pricePostNLMorning": "mypa-postnl-morning-price",
-            "pricePostNLFlatrate": "mypa-postnl-standard-price",
-            "pricePostNLEvening": "mypa-postnl-avond-price", /* yes, this one is dutch :( */
-        };
+        var deliveryDateId = mypajQuery('#mypa-select-date').val();
 
-        mypajQuery.each(priceMap, function (config, cssId) {
+        var currentDeliveryData = MyParcel.triggerDefaultOptionDelivery(deliveryDateId, deliveryMomentOfDay);
 
-            var price = 0;
+        if (currentDeliveryData !== null) {
+            currentDeliveryData.signed = MyParcel.DELIVERY_SIGNED;
+            mypajQuery('#mypa-input').val(JSON.stringify(currentDeliveryData));
+        }
+        if (typeof window.mypa != 'undefined') {
+            window.mypa.fn.fnCheckout.saveShippingMethod();
+        }
+    },
 
-            if (myParcelConfig[config]) {
-                price = parseFloat(myParcelConfig[config]);
-            }
+    triggerDefaultOptionDelivery: function (deliveryDateId, deliveryMomentOfDay) {
+        var dateArray = MyParcel.data.deliveryOptions.data.delivery[deliveryDateId];
+        var currentDeliveryData = null;
 
-            if (price > 0) {
-                mypajQuery('.' + cssId).html("+€ " + price);
+        mypajQuery.each(dateArray['time'], function(key, value) {
+            if (value.price_comment === deliveryMomentOfDay) {
+                currentDeliveryData = jQuery.extend({}, dateArray);
```

Comment:
Wat doet dit?

Reply:
Hiermee zouden alle eigenschappen van de `dateArray` en een andere array toegevoegd worden aan `currentDeliveryData` de null of undefined elementen zullen dan genegeerd worden.

Omdat er alleen gebruik wordt gemaakt van de `dateArray` hoe je de `jQuery.extend` niet te gebruiken.

Reply:
Done!